### PR TITLE
Fix transport state transition during close

### PR DESCRIPTION
**Pull request by SreeramGarlapati**
1. raise onError when service fails to send close frame and closes the transport; de-dup otherwise
2. handle double unbind of transport in graceful close cases

### CODE REVIEW

File: azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/amqp/ConnectionHandler.java [hunk 1]
```diff
@@ -101,11 +101,41 @@ public void onTransportError(Event event) {
...
+        // onTransportError event is not handled by the global IO Handler for cleanup
         transport.unbind();
     }
 
     @Override
+    public void onTransportClosed(Event event) {
+
+        final Connection connection = event.getConnection();
+        final Transport transport = event.getTransport();
+        final ErrorCondition condition = transport.getCondition();
```
Comment (JamesBirdsall) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/amqp/ConnectionHandler.java]: Could transport ever be null?

Comment (SreeramGarlapati) [inline azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/amqp/ConnectionHandler.java]: This is a transport event - so 'no'.

**Comment by SreeramGarlapati** [comment]
Closing and re-opening to test the appveyor system.

**Comment by SreeramGarlapati** [comment]
Closing and re-opening to test the appveyor system.

**Pull request merged**